daemon/daemon.go

```diff
@@ -77,25 +78,42 @@ type Command struct {
 	// is this path accessible on the snapd-snap socket?
 	SnapOK bool
 
+	// can polkit grant access?
+	ActionID string
+
 	d *Daemon
 }
 
+var polkitCheckAuthorization = polkit.CheckAuthorization
+
 func (c *Command) canAccess(r *http.Request, user *auth.UserState) bool {
 	if user != nil {
 		// Authenticated users do anything for now.
 		return true
 	}
 
 	isUser := false
-	uid, err := ucrednetGetUID(r.RemoteAddr)
+	pid, uid, err := ucrednetGet(r.RemoteAddr)
 	if err == nil {
 		if uid == 0 {
 			// Superuser does anything.
 			return true
 		}
 
+		if c.ActionID != "" {
+			subject := polkit.ProcessSubject{Pid: int(pid)}
+			if result, err := polkitCheckAuthorization(subject, c.ActionID, nil, polkit.CheckAuthorizationAllowUserInteraction); err == nil {
```

Comment:
I think we'll probably want to retain the flags argument, and let the client decide whether interaction should be allowed (e.g. if you run a snap command from a cron job, you probably don't want interactive auth).

The details map is not particularly useful with the version of polkit we currently ship in Ubuntu, but does offer some value with more recent releases.  If we protected the package install action with polkit and provided appropriate details, an administrator might be able to write a policy that prevented the user from installing snaps with classic confinement, for instance.